File: src/main/java/io/github/takenoha/towerdefense/domain/EnemyPathAction.java
package io.github.takenoha.towerdefense.domain;

/** Safe next-step choice for the role-aware enemy navigation planner. */
public enum EnemyPathAction {
    ADVANCE,
    BREAK_OBSTACLE,
    BUILD_SUPPORT,
    RECALCULATE_PATH,
    RECOVER
}

File: src/main/kotlin/io/github/takenoha/towerdefense/domain/EnemyPathAction.kt
package io.github.takenoha.towerdefense.domain

/** Safe next-step choice for the role-aware enemy navigation planner. */
enum class EnemyPathAction {
    ADVANCE,
    BREAK_OBSTACLE,
    BUILD_SUPPORT,
    RECALCULATE_PATH,
    RECOVER,
}
